.github/workflows/performance.yml

```diff
@@ -1,175 +1,151 @@
+      - name: Fetch Runner From Cache
+        uses: actions/cache@v2
+        id: cache
         with:
-          name: runner
-      - name: change permissions
+          path: ${{ env.RUNNER_CACHE_PATH }}
+          key: ${{ runner.os }}-${{ hashFiles('performance/runner/Cargo.toml')}}-${{ hashFiles('performance/runner/src/*') }}
+
+      - name: Move Runner
+        run: mv performance/runner/target/release/runner ./
+
+      - name: Change Runner Permissions
         run: chmod +x ./runner
-      - name: run
+
+      # `${{ github.workspace }}` is used to pass the absolute path
+      - name: Run Measurement
         run: ./runner measure -b dev -p ${{ github.workspace }}/performance/projects/
-      - uses: actions/upload-artifact@v2
-        with:
-          name: dev-results
-          path: performance/results/
 
-  # run the performance measurements on the release branch which we use
-  # as a performance baseline. This part takes by far the longest, so
-  # we do everything we can first so the job fails fast.
-  # -----
-  # we need to checkout dbt twice in this job: once for the baseline dbt
-  # version, and once to get the latest regression testing projects,
-  # metrics, and runner code from the develop or current branch so that
-  # the calculations match for both versions of dbt we are comparing.
-  measure-baseline:
-    needs: [build-runner]
-    name: Measure Baseline Branch
-    runs-on: ubuntu-latest
-    steps:
-      - name: checkout latest
-        uses: actions/checkout@v2
-        with:
-          ref: "0.20.latest"
-      - name: Setup Python
-        uses: actions/setup-python@v2.2.2
-        with:
-          python-version: "3.8"
-      - name: move repo up a level
-        run: mkdir ${{ github.workspace }}/../baseline/ && cp -r ${{ github.workspace }} ${{ github.workspace }}/../baseline
-      - name: "[debug] ls new dbt location"
-        run: ls ${{ github.workspace }}/../baseline/dbt/
-      # installation creates egg-links so we have to preserve source
-      - name: install dbt from new location
-        run: cd ${{ github.workspace }}/../baseline/dbt/ && pip install -r dev-requirements.txt -r editable-requirements.txt
-      # checkout the current branch to get all the target projects
-      # this deletes the old checked out code which is why we had to copy before
-      - name: checkout dev
-        uses: actions/checkout@v2
-      - name: install hyperfine
-        run: wget https://github.com/sharkdp/hyperfine/releases/download/v1.11.0/hyperfine_1.11.0_amd64.deb && sudo dpkg -i hyperfine_1.11.0_amd64.deb
-      - uses: actions/download-artifact@v2
-        with:
-          name: runner
-      - name: change permissions
-        run: chmod +x ./runner
-      - name: run runner
-        run: ./runner measure -b baseline -p ${{ github.workspace }}/performance/projects/
-      - uses: actions/upload-artifact@v2
+      - name: Upload Results
+        uses: actions/upload-artifact@v2
```

Comment:
Sweet! Confirming that we'll have this data saved to / available from S3?

Reply:
not at first. the first version of this test suite is going to add a commit to the repository with a json file for the full modeling of each release. The samples are printed in github actions but ultimately not saved.

If we want any of the above to be available in s3 we can track that work with a separate ticket.